software/msisensor/msi/main.nf

```diff
@@ -0,0 +1,41 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process MSISENSOR_MSI {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::msisensor=0.5" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/msisensor:0.5--hb3646a4_2"
+    } else {
+        container "quay.io/biocontainers/msisensor:0.5--hb3646a4_2"
+    }
+
+    input:
+    tuple val(meta), path(normal_bam), path(normal_bai), path(tumor_bam), path(tumor_bai), val(metascan), path(homopolymers)
+
+    output:
+    tuple val(meta), path("*.{output,output_dis,output_germline,output_somatic}"), emit: report
+
+    path "*.version.txt", emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    msisensor msi -d ${homopolymers} \\
+                    -n ${normal_bam} \\
+                    -t ${tumor_bam} \\
+                    -o ${prefix}.paired.output \\
+                    $options.args
```

Comment:
```suggestion
    output:
    tuple val(meta), path("${prefix}.output")         , emit: output
    tuple val(meta), path("${prefix}.output_dis")     , emit: output_dis
    tuple val(meta), path("${prefix}.output_germline"), emit: output_germline
    tuple val(meta), path("${prefix}.output_somatic") , emit: output_somatic
    path "*.version.txt"                              , emit: version

    script:
    def software = getSoftwareName(task.process)
    prefix       = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
    """
    msisensor \\
        msi \\
        -d $homopolymers \\
        -n $normal_bam \\
        -t $tumor_bam \\
        -o $prefix \\
        $options.args
```

Reply:
Some re-structuring to output individual `emit` for each file created and to only use `$prefix` to name the output files. This just makes it more flexible for developer to call their output files whatever they want :) 

If you want to call the output files this for the tests then you can use something like the below in the `main.nf` for the tests (see `suffix` declaration at the end):
https://github.com/nf-core/modules/blob/13bd67ed8aca271a260c78ca41afe9e1f5c217d7/tests/software/gatk4/variantfiltration/main.nf#L5